src/android/CodePushReportingManager.java

```diff
@@ -13,28 +12,7 @@
 
     private Activity cordovaActivity;
     private CodePushPreferences codePushPreferences;
-
-    /**
-     * Defines application statuses we use in reporting events from the native to the JS layer.
-     */
-    public static enum Status {
-        STORE_VERSION(0),
-        UPDATE_CONFIRMED(1),
-        UPDATE_ROLLED_BACK(2);
-
-        private int value;
-
-        Status(int i) {
-            this.value = i;
-        }
-
-        /**
-         * Returns the value associated with this enum.
-         */
-        public int getValue() {
-            return this.value;
-        }
-    }
+    private Boolean hasFailedReport = null;
```

Comment:
http://stackoverflow.com/questions/19148282/read-speed-of-sharedpreferences

they're in memory after the first read/write, so I wouldn't be worried about performance. ultimately it's fine as a variable, but if this behavior gets more complex I would definitely switch to using the prefs every time.